src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+
+		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][1].populate(gameSession.getGameBoard().getBlock(1),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][2].populate(gameSession.getGameBoard().getBlock(2),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[1][0].populate(gameSession.getGameBoard().getBlock(3),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][1].populate(gameSession.getGameBoard().getBlock(4),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][2].populate(gameSession.getGameBoard().getBlock(5),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[2][0].populate(gameSession.getGameBoard().getBlock(6),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][1].populate(gameSession.getGameBoard().getBlock(7),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][2].populate(gameSession.getGameBoard().getBlock(8),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+	}
+	
+	public void populatePencilMark(GameSession gameSession) {
+
+		this.gameSession = gameSession;
+
+		blocks[0][0].populatePencilMark(gameSession.getGameBoard().getBlock(0),
+				gameSession);
+		blocks[0][1].populatePencilMark(gameSession.getGameBoard().getBlock(1),
+				gameSession);
+		blocks[0][2].populatePencilMark(gameSession.getGameBoard().getBlock(2),
+				gameSession);
+
+		blocks[1][0].populatePencilMark(gameSession.getGameBoard().getBlock(3),
+				gameSession);
+		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
+				gameSession);
+		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
+				gameSession);
+
+		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
+				gameSession);
+		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
+				gameSession);
+		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
+				gameSession);
+	}
+
+	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
+		
+		if (null != selectedCell) {
+			/* existing selection, clear it */
+			selectedCell.unselect();			
+		}
+		
+		selectedCell = blocks[cellCoordinates.getBlockIndex()/3][cellCoordinates.getBlockIndex()%3].getSelectedCell(cellCoordinates);
+		selectedCell.updateSelectedCellFromHint(number);
+	}
+	
+	private FocusAdapter buildFocusAdapter() {
+
+		return new FocusAdapter() {
+			public void focusGained(FocusEvent focusEvent) {
+				cellGainedFocus(focusEvent);
+			}
+
+			public void focusLost(FocusEvent focusEvent) {
+				cellLostFocus(focusEvent);
+			}
+		};
+	}
+	
+	private void cellGainedFocus(FocusEvent focusEvent) {
+		CellGUI newSelectedCell = (CellGUI) ((JTextField) focusEvent.getSource())

```

Comment:
Will add more comments throughout the code.

On Sun, Jul 12, 2015 at 10:14 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34429671>:
>
> > +	
> > +	private FocusAdapter buildFocusAdapter() {
> > +
> > +		return new FocusAdapter() {
> > +			public void focusGained(FocusEvent focusEvent) {
> > +				cellGainedFocus(focusEvent);
> > +			}
> > +
> > +			public void focusLost(FocusEvent focusEvent) {
> > +				cellLostFocus(focusEvent);
> > +			}
> > +		};
> > +	}
> > +	
> > +	private void cellGainedFocus(FocusEvent focusEvent) {
> > +		CellGUI newSelectedCell = (CellGUI) ((JTextField) focusEvent.getSource())
>
> Can you add a comment explaining this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34429671>.
>